### ISSUE TITLE
:ruby break HAML-lint

### ISSUE THREAD

**Issue by Ana06** [brigade/haml-lint#287]
This code:

```ruby
- @project = @image.package.try(:project)
- @package = @image.package
```
produces the following offenses (we have some custom configuration, but it is just related to what files are inspected):

```ruby
app/views/webui2/webui/_tabs.html.haml:1 [W] InstanceVariables: Avoid using instance variables in not_breadcrumb_items views
app/views/webui2/webui/_tabs.html.haml:2 [W] InstanceVariables: Avoid using instance variables in not_breadcrumb_items views
```

While there are not offenses with these code:

```haml
:ruby
  @project = @image.package.try(:project)
  @package = @image.package
```

I don't really like the use of `:ruby`, but this looks like a bug to me, as it seems `:ruby` avoid that hamls inspects the rest of the code. :thinking:

**Comment by michaelherold** [comment]
Switching to using a `:ruby` block fixes the lints by happenstance. `:ruby` blocks are only linted via Rubocop (at least that's how I remember it), which means that the `InstanceVariables` linter isn't run on that code.

The true fix for those lints would be to pass `@image` in as a local into your partial. Your use of `@project` and `@package` shouldn't be using instance variables either and you can rename them to `project` and `package`.

**Comment by sds** [comment]
You must write your linter to explicitly lint `:ruby` filters. Unless you specify this, it will ignore them.

You can see an example of how to write one by looking at [`RubyExtractor`](https://github.com/brigade/haml-lint/blob/e86a20fdfbd1e940decb216608063ecd0397a920/lib/haml_lint/ruby_extractor.rb#L126)

I'm going to close this as the issue lies with the linter implementation, not HAML-Lint.
